packages/eslint-plugin/src/rules/ban-types.ts

```diff
@@ -82,32 +85,6 @@ const defaultTypes: Types = {
       'If you are expecting the function to accept certain arguments, you should explicitly define the function shape.',
     ].join('\n'),
   },
-
-  // object typing
-  Object: {
```

Comment:
We didn't explicitly discuss `Object`, but I think it makes sense in the spirit of this change to switch it to being treated as another banned uppercase alias. I've never seen people confuse `Object` for `{}`.